Initial package organization for garden builds

mjcarroll opened the pull request.

Commit:
Remove existing implementation

Signed-off-by: Michael Carroll <michael@openrobotics.org>

Commit:
Include initial package information for garden

Signed-off-by: Michael Carroll <michael@openrobotics.org>

Commit:
Update README and add LICENSE

Signed-off-by: Michael Carroll <michael@openrobotics.org>

mjcarroll commented on example/bazel.repos:
```diff
@@ -1,73 +1 @@
 repositories:
```
Yes, the CI and installation instructions make use of it to import the repos with the correct branches.

iche033 commented on README.md:
````diff
@@ -1,70 +1,89 @@
+
+```
+# Download bazelisk from Github releases
+wget https:://github.com/bazelbuild/bazelisk/releases/download/v1.16.0/bazelisk-linux-amd64
````
extra colon in `http:://`

mjcarroll replied:
Fixed.

Commit:
Clean up unused drake rules

Signed-off-by: Michael Carroll <michael@openrobotics.org>

iche033 reviewed (commented):
went through the README instructions and they worked fine for me

Commit:
Change license to match Drake and credit

Signed-off-by: Michael Carroll <michael@openrobotics.org>

mjcarroll commented:
It is generally going to be easier to keep the license BSD-3-Clause to match the upstream drake implementation I am using as the basis of our new implementation.  I have changed the license and added a credit in the readme to reflect this.

Commit:
Add caching and log artifacts to GitHub actions

Signed-off-by: Michael Carroll <michael@openrobotics.org>

nkoenig commented on README.md:
````diff
@@ -1,70 +1,89 @@
 
 ```
-wget https://raw.githubusercontent.com/ignitionrobotics/ign-bazel/master/example/bazel.repos
+wget https://raw.githubusercontent.com/gazebosim/gz-bazel/garden/example/bazel.repos
````
What file should I use? This link doesn't work, and I tried https://github.com/gazebosim/gz-bazel/pull/43/files#diff-b3c5fb2260c8de45c8de2a1b0170ffd91e9136b9e81a774595ce0be17dd1a374, but it's missing the `gz-bzel` repo and when I try to include it I get the following build errors:

```
ERROR: /home/nkoenig/gz/msgs/BUILD.bazel:94:17: no such package '@com_google_protobuf//': The repository '@com_google_protobuf' could not be resolved: Repository '@com_google_protobuf' is not defined and referenced by '//msgs:gzmsgs_cc_proto'
ERROR: /home/nkoenig/gz/msgs/BUILD.bazel:94:17: every rule of type _gz_msgs_generate_gen implicitly depends upon the target '@com_google_protobuf//:protoc', but this target could not be found because of: no such package '@com_google_protobuf//': The repository '@com_google_protobuf' could not be resolved: Repository '@com_google_protobuf' is not defined
ERROR: Analysis of target '//msgs:gzmsgs_cc_proto' failed; build aborted: Analysis failed
```

mjcarroll replied:
Ah, there is probably a bit of a chicken-and-egg problem here.

For now, I am cloning the `gz-bazel` repo separately to get the right branch, since it is the one changing the most frequently.  CI also assumes that `gz-bazel` is checked out separately (because actions will always do that) and then copied into the workspace.

This PR (and the sequence that follows it for each package) is intended to replace #43, that one can be ignored.

For now, do 

```
cd ~/gz
git clone https://github.com/gazebosim/gz-bazel bazel -b mjcarroll/init_garden
```

mjcarroll replied:
Additionally, this PR is only setting up the CI and the initial stuff we need to build the `bazel` package.  The ones on top actually build up the gazebo packages.

iche033 commented on README.md:
````diff
@@ -73,3 +92,52 @@ To test:
+bazel-bin/bazel/lint/buildifier //common/BUILD.bazel  # Only reformat one file.
+```
+
+If buildifier was not build with the workspace (unusual), but it via
````
not sure what this means? `but` -> `build` ?

mjcarroll replied:
Fixed, thanks.

iche033 reviewed (approved):
just one minor comment, otherwise looks good to me

nkoenig commented:
I'm getting the following errors when running `bazel build //...`

```
Starting local Bazel server and connecting to it...
ERROR: /home/nkoenig/gz/msgs/BUILD.bazel:94:17: no such package '@com_google_protobuf//': The repository '@com_google_protobuf' could not be resolved: Repository '@com_google_protobuf' is not defined and referenced by '//msgs:gzmsgs_cc_proto'
ERROR: /home/nkoenig/gz/msgs/BUILD.bazel:94:17: every rule of type _gz_msgs_generate_gen implicitly depends upon the target '@com_google_protobuf//:protoc', but this target could not be found because of: no such package '@com_google_protobuf//': The repository '@com_google_protobuf' could not be resolved: Repository '@com_google_protobuf' is not defined
ERROR: Analysis of target '//msgs:gzmsgs_cc_proto' failed; build aborted: Analysis failed
INFO: Elapsed time: 2.151s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (76 packages loaded, 838 targets c\
onfigured)
```

mjcarroll commented:
> ERROR: Analysis of target '//msgs:gzmsgs_cc_proto' failed; build aborted: Analysis failed

For this PR, you shouldn't have `msgs`.  The only repo here should be `gz/bazel`

Commit:
Readme tweak

Signed-off-by: Michael Carroll <michael@openrobotics.org>

nkoenig commented:
> I'm getting the following errors when running `bazel build //...`
> 
> ```
> Starting local Bazel server and connecting to it...
> ERROR: /home/nkoenig/gz/msgs/BUILD.bazel:94:17: no such package '@com_google_protobuf//': The repository '@com_google_protobuf' could not be resolved: Repository '@com_google_protobuf' is not defined and referenced by '//msgs:gzmsgs_cc_proto'
> ERROR: /home/nkoenig/gz/msgs/BUILD.bazel:94:17: every rule of type _gz_msgs_generate_gen implicitly depends upon the target '@com_google_protobuf//:protoc', but this target could not be found because of: no such package '@com_google_protobuf//': The repository '@com_google_protobuf' could not be resolved: Repository '@com_google_protobuf' is not defined
> ERROR: Analysis of target '//msgs:gzmsgs_cc_proto' failed; build aborted: Analysis failed
> INFO: Elapsed time: 2.151s
> INFO: 0 processes.
> FAILED: Build did NOT complete successfully (76 packages loaded, 838 targets c\
> onfigured)
> ```

Ahh, I see. thanks.
